src/System.Security.Cryptography.Encoding/src/System/Security/Cryptography/Base64Transforms.cs

```diff
@@ -96,135 +103,217 @@ public class FromBase64Transform : ICryptoTransform
     {
         private byte[] _inputBuffer = new byte[4];
         private int _inputIndex;
-        private FromBase64TransformMode _whitespaces;
+        private readonly FromBase64TransformMode _whitespaces;
 
         public FromBase64Transform() : this(FromBase64TransformMode.IgnoreWhiteSpaces) { }
         public FromBase64Transform(FromBase64TransformMode whitespaces)
         {
             _whitespaces = whitespaces;
-            _inputIndex = 0;
         }
 
         // Converting from Base64 generates 3 bytes output from each 4 bytes input block
-        public int InputBlockSize => 1;
+        public int InputBlockSize => 4;
```

Comment:
Previously it was `1`. According to the algorithm (base64) `4` is correct, but was it intentional or just a failure, that got ported over from .NET Full?

Reply:
I don't see any reason from the .NET Framework implementation why this should say 4 (the implementation seems to accept any number of bytes, and correctly handles up to 4 non-whitespace ASCII characters).

Those semantics need to be preserved with this change.

Reply:
For base64 decode the minimum input size is 4. 
Why should it be set to 1? This would result in more iterations over the source-data to transform it.

It's correct that the implementation can handle inputs < 4, as they get stored in the `_inputBuffer` and are reused later.

Additional rationale for the change: https://github.com/dotnet/corefx/pull/38062/files#r290872023

Reply:
Since this is a publicly exposed value, it needs to be restored to 1 for inclusion in .NET Core 3.0 (just not comfortable taking it this late).  If that invalidates this change then we can just hold the change until master becomes vNext.

Reply:
I've restored it to 1 and updated the benchmark [results](https://github.com/dotnet/corefx/pull/38062#issuecomment-504578089).

---

For me the `1` is a "bug" in .NET Full (hence also in .NET Core) and should be fixed, i.e. set to `4`.
(note: bug is a slightly hard word here, as it is not a functional bug, rather a too small value set)

What is the correct procedure to file an issue for this?
Or is the `1` fixed, so no issue is needed?

Reply:
> What is the correct procedure to file an issue for this?

Just open an issue as normal; but the fix won't be accepted until master is no longer flowing to 3.0.  The problem is that it's such a low level change that probably doesn't, but might, matter that I'm just concerned that there's aren't enough previews left for someone to find a reason why it had to be 1 and us to fix it.  

6 months ago, I'd have been hesitant, but since I can't think of any specific reason why I'd let it go through after making sure we had tests for things like using a stream that only allows 3 bytes to be read at a time to be processed via CryptoStream with this transform to make sure that that layer was working successfully.

Reply:
Thanks for the info.

I'll file an issue after this PR is finished and master is open for "vnext".

Reply:
-> https://github.com/dotnet/corefx/issues/39583